src/tensors/abstracttensor.jl

```diff
@@ -497,7 +497,7 @@ function Base.convert(::Type{Array}, t::AbstractTensorMap)
     else
         cod = codomain(t)
         dom = domain(t)
-        T = isreal(sectorscalartype(I)) ? scalartype(t) : complex(scalartype(t))
+        T = sectorscalartype(I) <: Complex ? complex(scalartype(t)) : scalartype(t)
```

Comment:
Given that `real` and `complex` do work on `Number` subtypes to e.g. convert Float64 to ComplexF64, I understand the confusion of `isreal` not working. Maybe that is worthy of a Julia issue.

Reply:
Yeah, it's not super consistent. I just realised I still messed up with the integer types, life is hard